### PR TITLE
Feature/connect profile

### PR DESCRIPTION
**Pull request by brian-glovebox**
### What this PR does (required):
- Able to get contests and user information from the backend and display on the profile page, also added a link in the authmenu to direct to the profile page, and wrote a few front end api calls in the helpers folder

### Screenshots / Videos (required):
Screen Shot 2021-07-15 at 10 20 55 AM

### Any information needed to test this feature (required):
- Sign in as user and direct to profile page, if you have contests create, they should display, if there are none, you will see a message displayed "None Available". Make sure you have created a contest under the user you are logged in as.

### CODE REVIEW

File: client/src/components/ContestList/ContestList.tsx [hunk 1]
```diff
@@ -6,37 +6,52 @@ import Avatar from '@material-ui/core/Avatar';
...
-                </ListItem>
-            </List>
+            {/* Write ternary operator checking to see if any data has been recieved, otherwise return default message */}
+            {userContests.length > 0 ? (
+
+                <List className={classes.list}>
+                    {/* map through given data and alter the return inside of the map */}
+                    {userContests.map(contest => {
+                        return (
+                            <ListItem key={contest.userId}>
```
Comment (rajivtitus) [inline client/src/components/ContestList/ContestList.tsx]: Please use a different key here, as the `userId` will be the same for every item.

File: client/src/interface/AuthApiData.ts [hunk 2]
```diff
@@ -9,4 +10,11 @@ export interface AuthApiDataSuccess {
 export interface AuthApiData {
   error?: { message: string };
   success?: AuthApiDataSuccess;
+  contests?: [Contest]
 }
+
+export interface AuthApiId {
```
Comment (rajivtitus) [inline client/src/interface/AuthApiData.ts]: Maybe rename this interface (as we are using it for contests and not auth)?  :)

File: client/src/pages/Profile/Profile.tsx [hunk 3]
```diff
@@ -41,6 +44,42 @@ export default function Profile(): JSX.Element {
         index: number;
     }
 
+    useEffect(() => {
+        async function getUserContests() {
+            const userContests = await getContestByUser();
```
Comment (rajivtitus) [inline client/src/pages/Profile/Profile.tsx]: Maybe handle errors here too? :)

**Review by rajivtitus — APPROVED** [review]
Good work!

Just 2-3 comments, please have a look.

File: client/src/components/ContestList/ContestList.tsx (hunk 1 above)
Comment (brian-glovebox) [reply client/src/components/ContestList/ContestList.tsx]: good catch, thank you, will be updating changes shortly and merging.

**Pull request merged**